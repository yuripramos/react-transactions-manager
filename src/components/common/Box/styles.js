import styled, { css } from "styled-components";
import { white, grey10 } from "../../../styles/settings";
import { rem, media } from "../../../styles/tools/index";
import { isResponsive } from "../../../utils/getResolution";

export const Title = styled.span`
  color: ${grey10};
  font-size: ${rem(11)};
  font-weight: bold;
  line-height: 1.45;
  text-transform: uppercase;
`;

export const IconContainer = styled.span`
  align-items: center;
  background: ${white};
  border-radius: ${rem(4)};
  color: ${grey10};
  display: inline-flex;
  height: ${rem(24)};
  justify-content: center;
  margin-right: ${rem(9)};
  width: ${rem(24)};
`;

export const Header = styled.div`
  align-items: center;
  display: flex;
`;

export const Wrapper = styled.div`
  display:inline-flex;
  flex-wrap: wrap;
  ${({ noTitle }) =>
  noTitle &&
    css`
      ${media.xs(css`
        transform: translateY(0px);
      `)};
      ${media.md(css`
        transform: translateY(10px);
      `)};
  `};
  ${({ width }) =>
  width &&
  css`
    width: ${isResponsive() ? rem(width/0.7) : rem(width)};
  `};
`;

