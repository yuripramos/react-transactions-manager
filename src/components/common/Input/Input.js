import React, { Component } from "react";
import { string, func, bool, array, number, oneOfType } from "prop-types";

/* eslint-disable */

import {
  InputWrapper,
  InputField,
  Label,
  IconContainer,
  ValidityContainer,
  Tooltip,
  Prefix
} from "./styles";
import Icon from "../Icon";

export const KEYCODES = {
  ENTER: 13,
  BACKSPACE: 8,
  ALL: "all"
};

class Input extends Component {
  constructor(props) {
    super(props);
    this.onKeyDown = this._onKeyDown.bind(this);
    this.state = {
      isFocused: false,
      emptyState: false,
      isValid: typeof props.valid === "boolean" ? props.valid : null
    };
  }

  componentDidUpdate(prevProps) {
    if (this.props.value === "" && this.state.isValid !== null) {
      this.setState({
        isValid: null
      });
    }

    if (prevProps.value === "" && this.props.value.length > 4) {
      this.setState({
        isValid:
          this.props.value === ""
            ? null
            : typeof this.props.valid === "function"
            ? this.props.valid()
            : this.state.isValid
      });
    }
  }

  _onKeyDown(event) {
    const { disableKeys, onKeyDown } = this.props;

    const disableKeysArray = Array.isArray(disableKeys)
      ? disableKeys
      : [disableKeys];

    const disabledKey = disableKeysArray.find(
      disableKey => event.keyCode == disableKey || event.which == disableKey
    );

    if (disableKeysArray.includes(KEYCODES.ALL) || disabledKey) {
      event.preventDefault();
      return;
    }
    onKeyDown && onKeyDown(event);
  }

  render() {
    const {
      label,
      value,
      placeholder,
      onChange,
      onBlur,
      onFocus,
      onKeyUp,
      type,
      name,
      icon,
      valid,
      tinyLabels,
      toolTip,
      dataTest,
      width,
      maskType,
      refType,
      prefix,
      maxLength,
      ...props
    } = this.props;
    const { isFocused, emptyState, isValid } = this.state;
    return (
      <InputWrapper valid={emptyState ? false : isValid} width={width}>
        {prefix && (value || isFocused) && <Prefix>{prefix}</Prefix>}
        <InputField
          {...props}
          required
          maxLength={maxLength}
          onChange={e => {
            onChange && onChange(e);
            this.setState({ emptyState: e.target.value === "" });
          }}
          innerRef={this.props.innerRef}
          onFocus={e => {
            onFocus && onFocus(e);
            (prefix || toolTip) && this.setState({ isFocused: true });
          }}
          onBlur={e => {
            onBlur && onBlur(e);
            this.setState({
              isFocused: false,
              isValid: valid ? valid() : this.state.isValid
            });
          }}
          prefix={prefix}
          type={type}
          value={value}
          name={name}
          onKeyDown={this.onKeyDown}
          autoComplete="new-password"
          placeholder={placeholder}
          data-test={dataTest ? dataTest : Input.displayName}
          isEmpty={!value}
          tinyLabels={tinyLabels}
        />
        {label && <Label tinyLabels={tinyLabels}>{label}</Label>}
        {/* {(isValid !== null || emptyState) && !disabled && (
          <ValidityContainer>
            {isValid && !emptyState ? (
              <Icon type="Check" width="13" height="13" />
            ) : (
              <Icon type="Attention" width="24" height="24" />
            )}
          </ValidityContainer>
        )} */}
      </InputWrapper>
    );
  }
}

Input.displayName = "Input";

Input.defaultProps = {
  placeholder: "",
  type: "text",
  name: "",
  label: null,
  icon: null,
  valid: () => {},
  onFocus: null,
  tinyLabels: null,
  onBlur: () => {},
  value: "",
  disableKeys: []
};

Input.propTypes = {
  placeholder: string,
  type: string,
  name: string,
  label: string,
  icon: string,
  tinyLabels: bool,
  value: oneOfType([number, string]),
  valid: oneOfType([bool, func]),
  onBlur: func,
  onFocus: func,
  disableKeys: oneOfType([number, string, array]),
  onChange: func
};

export default Input;
