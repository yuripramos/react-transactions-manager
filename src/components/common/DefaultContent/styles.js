import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { black30, black50 } from "../../../styles/settings";

export const Wrapper = styled.div`
  text-align: center;
  padding: ${rem(25)} ${rem(10)};

  ${media.md(css`
    padding: ${rem(50)} ${rem(10)};
  `)};
`;

export const IconWrapper = styled.div`
  margin-bottom: ${rem(11)};

  ${media.md(css`
    margin-bottom: ${rem(15)};
  `)};
`;

export const PrimaryText = styled.h3`
  font-size: ${rem(16)};
  letter-spacing: ${rem(0.5)};
  text-align: left;
  color: ${black30};
  margin-bottom: ${rem(10)};
  display: inline-block;
  text-align: center;

  ${media.md(css`
    font-size: ${rem(28)};
    margin-bottom: ${rem(20)};
  `)};
`;
export const SecondaryText = styled.p`
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.6)};
  color: ${black50};
  line-height: 1.45;
  text-align: center;

  ${media.md(css`
    font-size: ${rem(22)};
    line-height: 1.45;
  `)};
`;
