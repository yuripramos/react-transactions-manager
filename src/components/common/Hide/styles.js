import styled, { css } from "styled-components";
import { media } from "../../../styles/tools";

export const HideStyle = styled.div`
  display: none;

  ${({ below }) =>
    below &&
    media[below](css`
      display: block;
    `)};

  ${({ above }) =>
    above &&
    css`
      display: block;

      ${media[above](css`
        display: none;
      `)};
    `};
`;
