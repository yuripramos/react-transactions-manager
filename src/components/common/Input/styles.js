import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";

import {
  white,
  grey30,
  black50,
  darkBlue,
  lightestBlue,
  red,
  grey180
} from "../../../styles/settings";

export const IconContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 0;
  width: ${rem(55)};
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translateY(-50%);
  color: ${black50};
  transition: 0.3s color;
  pointer-events: none;
`;

export const Prefix = styled.span`
  position: absolute;
  padding: ${rem(15)} ${rem(0)} ${rem(0)} ${rem(10)};
  pointer-events: none;
  font-family: Lato;
  font-size: ${rem(13)};
  color: ${grey180};
`;

export const Tooltip = styled.div`
  visibility: visible;

  ${({ hide }) =>
    hide &&
    css`
      visibility: hidden;
    `};
`;

export const Label = styled.label`
  position: absolute;
  left: ${rem(52)};
  top: 50%;
  transform: translateY(-50%);
  pointer-events: none;
  transition: 0.3s top, 0.2s color, 0.3s font-size;
  font-family: Lato;
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.1)};
  color: ${black50};
  ${({ tinyLabels }) =>
    tinyLabels &&
    css`
      font-size: ${rem(10)};
      left: ${rem(10)};
      top: 50%;
      transform: translateY(-50%);
    `};
`;

const revealContent = css`
  ~ ${Label} {
    top: ${rem(22)};
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent2 = css`
  ~ ${Label} {
    top: ${rem(10)};
    font-family: Lato Bold;
    font-size: ${rem(9)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent3 = css`

  ~ ${Label} {
    top: ${rem(12)};
    color: ${black50};
  }
`;

export const InputField = styled.input`
  width: 100%;
  height: ${rem(46)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.1)};
  color: ${black50};
  padding: 0 ${rem(52)};
  transition: 0.3s padding, 0.3s border;
  box-shadow: none;

  ::-ms-clear {
    display: none !important;
  }
  ${({ prefix }) =>
    prefix &&
    css`
      padding-left: ${rem(32)} !important;
    `};

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }

  ${({ tinyLabels }) =>
    tinyLabels &&
    css`
      padding: 0 ${rem(9.2)};
    `};

  :focus {
    ${({ tinyLabels }) => (tinyLabels ? revealContent2 : revealContent)};
  }
  ${({ isEmpty }) => !isEmpty && revealContent};
  ${({ isEmpty, tinyLabels }) => !isEmpty && tinyLabels && revealContent2};
  :not(:focus) {
    ${({ tinyLabels, isEmpty }) => tinyLabels && !isEmpty && revealContent3};
  }
`;

export const InputWrapper = styled.div`
  width: 100%;
  display: inline-block;
  position: relative;
  height: ${rem(38)};
  margin-right: ${rem(25)};
  input:not(:focus) {
    ${({ valid }) =>
      valid === false &&
      css`
        border-color: ${red};
      `};
  }
  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};

  ~ input {
    background-color: #ccc;
  }
`;

export const WrapperDayPicker = styled.div`
  margin-right: ${rem(10)};
  .DayPickerInput > input {
    width: ${rem(60)};
    height: ${rem(46)};
    border-radius: ${rem(4)};
    background-color: ${white};
    border: solid ${rem(1)} ${grey30};
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    color: ${black50};
    padding: 0 ${rem(20)};
    transition: 0.3s padding, 0.3s border;
    box-shadow: none;

    ::-ms-clear {
      display: none !important;
    }

    :focus {
      outline: none;
      border: solid ${rem(1)} ${darkBlue};
    }
  }

  .DayPickerInput-OverlayWrapper {
    position: absolute;
    background: ${lightestBlue};
    left: 78px;
    top: -15px;
    ${media.md(css`
      left: 166px;
       top: -30px;
    `)};
  }

  .DayPickerInput-Overlay {
    position: absolute;
    left: 0;
    z-index: 999;

    background: ${lightestBlue};
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
  }
`;
