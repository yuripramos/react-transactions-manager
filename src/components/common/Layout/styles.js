import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";

export const Content = styled.div`
  ${media.sm(css`
    min-height: calc(100vh - ${rem(150)});
  `)};
  ${media.md(css`
    min-height: calc(100vh - ${rem(180)});
  `)};
  ${media.lg(css`
    min-height: calc(100vh - ${rem(173)});
  `)};
`;
