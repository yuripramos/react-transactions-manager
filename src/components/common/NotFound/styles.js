import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";

export const Content = styled.div`
  margin-top: ${rem(33)};

  a {
    width: auto;
  }

  ${media.md(css`
    margin-top: ${rem(50)};
  `)};
`;

export const Wrapper = styled.div`
  svg {
    height: ${rem(75)};
    width: auto;

    ${media.md(css`
      height: auto;
    `)};
  }
`;
