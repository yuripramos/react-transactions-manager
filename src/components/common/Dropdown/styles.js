import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";
import {
  white,
  grey30,
  black50,
  darkBlue,
  green,
  red
} from "../../../styles/settings";
import { Label } from "../../common/Input/styles";

export const DropdownField = styled.select`
  width: 100%;
  height: ${rem(48)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  font-family: Lato;
  font-size: ${rem(13)};
  letter-spacing: ${rem(0.8)};
  color: ${black50};
  padding: ${rem(15)} ${rem(55)} 0 ${rem(6.4)};
  transition: 0.3s padding, 0.3s border;
  box-shadow: none;

  ::-ms-clear {
    display: none !important;
  }

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }

  ${({ isEmpty, tinyLabels }) => !isEmpty && tinyLabels && revealContent2};
  :not(:focus) {
    ${({ tinyLabels, isEmpty }) => tinyLabels && !isEmpty && revealContent3};
  }

  ${({ hasLabel }) =>
    !hasLabel &&
    css`
      height: auto;
      padding: ${rem(8)} 0 ${rem(8)} ${rem(15)};
      border-radius: ${rem(20)};
      appearance: none;
    `};
`;

export const DropdownWrapper = styled.div`
  letter-spacing: ${rem(0.1)};
  display: inline-block;
  position: relative;
  margin: 0 ${rem(6)} ${rem(6)} 0;

  select:not(:focus) {
    ${({ valid }) =>
      valid === false &&
      css`
        border-color: ${red};
      `};
  }

  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};
`;

const revealContent2 = css`
  ~ ${Label} {
    top: ${rem(10)};
    font-family: Lato Bold;
    font-size: ${rem(11)};
    letter-spacing: ${rem(0.1)};
    text-align: left;
    color: ${darkBlue};
  }
`;

const revealContent3 = css`
  ~ ${Label} {
    top: ${rem(12)};
    color: ${black50};
  }
`;

export const IconContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 0;
  width: ${rem(55)};
  display: flex;
  justify-content: center;
  align-items: center;
  transform: translateY(-50%);
  color: ${black50};
  transition: 0.3s color;
  pointer-events: none;
`;
