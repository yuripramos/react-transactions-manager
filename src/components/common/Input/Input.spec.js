import React from "react";
import { shallow } from "enzyme";

import Input from "./Input";
import { InputField, Tooltip } from "./styles";

describe("Input component", () => {
  it("should match snapshot", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with label", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      label: "label"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with success validation", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: true
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with icon", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: true,
      icon: "Icon"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with error validation", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: false
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with no validation", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with currency mask", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null,
      maskType: "currency"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });
  it("should match snapshot with prefix", () => {
    const props = {
      onChange: () => {},
      type: "text",
      name: "input",
      placeholder: "My input",
      prefix: "$"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });
  it("should match snapshot with datetime mask", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null,
      maskType: "datetime"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with datetime mask", () => {
    const props = {
      onChange: () => {},
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null,
      maskType: "cpfcnpj"
    };
    expect(shallow(<Input {...props} />)).toMatchSnapshot();
  });

  it("should trigger onKeyDown prop with event", () => {
    const props = {
      onChange: () => {},
      onKeyDown: jest.fn(),
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null
    };

    const input = shallow(<Input {...props} />);
    input.find(InputField).simulate("keydown", { target: "sample" });

    expect(props.onKeyDown).toHaveBeenCalledWith({ target: "sample" });
  });

  it("should prevent default if disabled key is clicked", () => {
    const props = {
      onChange: jest.fn(),
      onKeyDown: jest.fn(),
      type: "submit",
      name: "input",
      placeholder: "My input",
      valid: null,
      disableKeys: [13]
    };

    const input = shallow(<Input {...props} />);

    const preventDefault = jest.fn();

    input.find(InputField).simulate("keydown", { keyCode: 13, preventDefault });

    expect(preventDefault).toHaveBeenCalled();

    input.find(InputField).simulate("keydown", { which: 13, preventDefault });

    expect(preventDefault).toHaveBeenCalledTimes(2);
  });

  describe("_onKeyDown", () => {
    it("should invoke onKeyDown when keyCode is different then disableKeys", () => {
      const props = {
        onChange: jest.fn(),
        onKeyDown: jest.fn(),
        type: "submit",
        name: "input",
        placeholder: "My input",
        valid: null,
        disableKeys: [13]
      };

      const e = {
        keyCode: 1,
        which: 1,
        preventDefault: jest.fn()
      };

      const instance = shallow(<Input {...props} />).instance();

      instance._onKeyDown(e);

      expect(props.onKeyDown).toHaveBeenCalledTimes(1);
    });

    it("should invoke onKeyDown when keyCode is different then disableKeys", () => {
      const props = {
        onChange: jest.fn(),
        onKeyDown: jest.fn(),
        type: "submit",
        name: "input",
        placeholder: "My input",
        valid: null,
        disableKeys: [13]
      };

      const e = {
        keyCode: 13,
        which: 13,
        preventDefault: jest.fn()
      };

      const instance = shallow(<Input {...props} />).instance();

      instance._onKeyDown(e);

      expect(props.onKeyDown).not.toHaveBeenCalled();
    });
  });
  describe("inside components", () => {
    it("tooltip", () => {
      const props = {
        onChange: () => {},
        type: "submit",
        name: "input",
        placeholder: "My input",
        valid: null,
        toolTip: jest.fn()
      };
      const wrapper = shallow(<Input {...props} />);
      expect(wrapper.find(Tooltip).length).toBe(1);
    });
  });
});
