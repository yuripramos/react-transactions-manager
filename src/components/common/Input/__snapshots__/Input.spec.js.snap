// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshot 1`] = `
<styled.div
  valid={null}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
</styled.div>
`;

exports[`Input component should match snapshot with currency mask 1`] = `
<styled.div
  valid={null}
>
  <t
    guide={false}
    mask={false}
    onChange={[Function]}
    render={[Function]}
  />
</styled.div>
`;

exports[`Input component should match snapshot with datetime mask 1`] = `
<styled.div
  valid={null}
>
  <t
    guide={false}
    mask={
      Array [
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    render={[Function]}
  />
</styled.div>
`;

exports[`Input component should match snapshot with datetime mask 2`] = `
<styled.div
  valid={null}
>
  <t
    guide={false}
    mask={
      Array [
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        ".",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        "-",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    render={[Function]}
  />
</styled.div>
`;

exports[`Input component should match snapshot with error validation 1`] = `
<styled.div
  valid={false}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
  <styled.div>
    <Icon
      color={null}
      cursor={false}
      fixCrossAxisAlign={null}
      height="24"
      type="Attention"
      width="24"
    />
  </styled.div>
</styled.div>
`;

exports[`Input component should match snapshot with icon 1`] = `
<styled.div
  valid={true}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
  <styled.div>
    <Icon
      color={null}
      cursor={false}
      fixCrossAxisAlign={null}
      height={null}
      type="Icon"
      width={null}
    />
  </styled.div>
  <styled.div>
    <Icon
      color={null}
      cursor={false}
      fixCrossAxisAlign={null}
      height="13"
      type="Check"
      width="13"
    />
  </styled.div>
</styled.div>
`;

exports[`Input component should match snapshot with label 1`] = `
<styled.div
  valid={null}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder=""
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
  <styled.label
    tinyLabels={null}
  >
    label
  </styled.label>
</styled.div>
`;

exports[`Input component should match snapshot with no validation 1`] = `
<styled.div
  valid={null}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
</styled.div>
`;

exports[`Input component should match snapshot with prefix 1`] = `
<styled.div
  valid={null}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    prefix="$"
    required={true}
    tinyLabels={null}
    type="text"
    value=""
  />
</styled.div>
`;

exports[`Input component should match snapshot with success validation 1`] = `
<styled.div
  valid={true}
>
  <styled.input
    autoComplete="new-password"
    data-test="Input"
    disableKeys={Array []}
    disabled={false}
    isEmpty={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="My input"
    required={true}
    tinyLabels={null}
    type="submit"
    value=""
  />
  <styled.div>
    <Icon
      color={null}
      cursor={false}
      fixCrossAxisAlign={null}
      height="13"
      type="Check"
      width="13"
    />
  </styled.div>
</styled.div>
`;
